src/MooLite/core/inventory/Inventory.ts

```diff
@@ -87,23 +94,76 @@ export class Inventory {
         });
     }
 
-    updateCharacterDrink(actionTypeDrinkSlotsMap: Record<ActionTypeHrid, CharacterConsumable[]>) {
-        this._actionTypeDrinkSlotsMap = actionTypeDrinkSlotsMap;
+    updateCharacterDrinks(
+        actionTypeDrinkSlotsMap: Record<ActionTypeHrid, (CharacterConsumable | null)[]>,
+        notify: boolean = true
+    ) {
+        for (const actionType in actionTypeDrinkSlotsMap) {
+            if (!notify) {
+                return;
+            }
+            const consumables = actionTypeDrinkSlotsMap[actionType];
+
+            if (this._characterDrinks[actionType] == null) {
+                this._characterDrinks[actionType] = consumables;
+            }
+
+            this._characterDrinks[actionType].forEach((storedConsumable, i) => {
+                if (storedConsumable === null) {
+                    return;
+                }
+                const newConsumable = actionTypeDrinkSlotsMap[actionType][i];
+
+                // If stored consumable state is active and new state = inactive,
+                // dispatch the event. Checking for the difference stops the notification
+                // from being sent out every time updateCharacterDrink is called.
+                if (!newConsumable?.isActive && storedConsumable.isActive) {
+                    this._onConsumableDepleted.dispatch(storedConsumable);
+                }
+            });
+
+            // Save the new consumable slots state
+            this._characterDrinks[actionType] = actionTypeDrinkSlotsMap[actionType];
+        }
     }
 
-    updateCharacterFood(actionTypeFoodSlotsMap: Record<ActionTypeHrid, CharacterConsumable[]>) {
-        this._actionTypeFoodSlotsMap = actionTypeFoodSlotsMap;
+    updateCharacterFood(actionTypeFoodSlotsMap: Record<ActionTypeHrid, CharacterConsumable[]>, notify: boolean = true) {
+        for (const actionType in actionTypeFoodSlotsMap) {
+            if (!notify) {
+                return;
+            }
```

Comment:
Notify should still update the state, just not dispatch the event!

Reply:
Good point - I added it as another part of the `&&` statement - if this is a bit too messy i can just move the `if (notify)` statement inside the `forEach`